### DISCUSSION TITLE
AliasEditor (two questions)

### DISCUSSION THREAD

**Discussion by klaidoshka** [Zrips/CMI#4613] [Q&A]
**Questions:**

1) Is it possible to get custom error-message for not having certain permission for the aliased command? If not, could it be maybe added in the future? Or not really? ☹
2) 'MyCommand' (from which I would certainly switch over to CMI aliaseditor, if...) had "ADD_PERMISSION" commands. For example: Don't proceed with the certain command if player doesn't have certain permission. Is it possible to do so with CMI's aliaseditor? Or not really? Maybe it can be added in the future? Haha... 😅

What I mean by ADD_Permission command type:

```
sell:
  Cmds:
  - perm:use.sell?#! asPlayer! /sell $1-
```
At the moment such a thing just makes a loop (I believe?) and throws errors into the console (yup, saw them... 😫)

Not all of our plugins have a way for doing something like `/<plugin>:sell` :'/

**Comment by Zrips** [comment]
1. Currently it will only show generic message about missing permission node, but maybe we could add option to define custom one
2. Well your current usage does create loop as it simply says that after performing /sell you should be directed to /sell with maybe some variables you provided. This might need some readjustment from my side, tho its getting some what complicated to consider all possible variations of custom alias which should direct to correct place with all possible variable variations.

**Comment by CloudeLecaw** [comment]
Zrips said it very often now, you should always use the direct command instead of alais inside the alias to prevent issues like this.

In your case you orders your new /sell to execute itself what ends inna loop. You need to use the command behind /sell what would be /cmi sell here.

Players need the cmi.command.sell permission for this allready.. dont forget this ;)
Looking inside the wiki command struktur of /cmi sell tells us it only has one variable after sell. So its enough to use $1 instead of $1-
With just $1 it also work fine if player wrote /sell hand my sword
Than it execute /sell hand

But with your sell you can define a meassage like that allready - before the line with the breakup command simply add

```
sell:
  Cmds:
  - perm:use.sell@! msg! [playerName] You need permission use.sell and cmi.command.sell to execute the command /sell.
  - perm:use.sell?#! asPlayer! /cmi sell $1
```

**Comment by klaidoshka** [comment]
> 
> 
> Zrips said it very often now, you should always use the direct command instead of alais inside the alias to prevent issues like this.
> 
> In your case you orders your new /sell to execute itself what ends inna loop. You need to use the command behind /sell what would be /cmi sell here.
> 
> Players need the cmi.command.sell permission for this allready.. dont forget this ;)
> Looking inside the wiki command struktur of /cmi sell tells us it only has one variable after sell. So its enough to use $1 instead of $1-
> With just $1 it also work fine if player wrote /sell hand my sword
> Than it execute /sell hand
> 
> But with your sell you can define a meassage like that allready - before the line with the breakup command simply add
> 
> ```
> sell:
>   Cmds:
>   - perm:use.sell@! msg! [playerName] You need permission use.sell and cmi.command.sell to execute the command /sell.
>   - perm:use.sell?#! asPlayer! /cmi sell $1
> ```

Nah, I know how it works, just saying, that we're using another plugin and we got even more custom plugins, so later on we can't even use formats like `[ /<plugin_name>:<command>]`, the only way is to use the command itself, which would be already made in CMI to check the permission and give out custom error message or something like that (that's the feature I'd love to get =]), so in the end... It won't work? Because it'll create loop.

So, for me, yeah, would be nice to get overlay thing (to do something before using the command) or(and) custom-error message**s**.

Though I'm not really sure yet how these optional things like (@)(?)(#) work, but if you say it'd work **ONLY** when they don't have permission, I'll try it today! <3

**Comment by CloudeLecaw** [comment]
Here u see the example from the wiki:
`
If you want to check opposite condition, then use @ For example perm:cmi.testperm@! cmi heal [playerName] this will only perform command if player doesn’t have cmi.testperm permission node`

Description and examples u find here
https://www.zrips.net/cmi/commands/specialized/

How the symbols work:
- @ - means only do it when condition is not true
- ? - mean inform the player if condition is not true (these informing are generated from cmi, so it may not is what your looking for
- ‘#' means break up everthing (also all commands after, if your having more then one command under this condition
You can combine the conditions by using them after each other, like condition@?#!
Als conditions end allways with a ! At the end.

```
sell:
  Cmds:
  - perm:use.sell@! msg! [playerName] You need permission use.sell and cmi.command.sell to execute the command /sell.
  - perm:use.sell?#! asPlayer! /cmi sell $1
```
What this example would do step by step:
1. It checks if you have the permission use.sell
If you dont have it it prints out a meassage
2. It checks if you have the permission use.sell
If you have it it executes the command behind, if you dont have it breaks up and the process is over.

About your other plugin. its not good that you cant execute the command over an other way then just /sell.
All i can recommend here is disable sell in alias.yml and give this custom alias an other name. It looks like you have no Option if you need to execute /sell
Like u noticed allready a custom alais that execute same command as an other plugin should will create an infinite loop

**Comment by klaidoshka** [comment]
Wow, thanks Cloude! =]
A lot better explanation of them here than in wiki 🥰!
Yeah, that would work like "custom-error" message for me, no need for anything different.

But still! Overlay would be a thing 😅

However, if there is no way that Zrips would create new feature like this, I'll try to think of anything different for my commands, will not bother you =]!

If you want, you can close the thread. 🙂
